echo-api/src/main/java/com/netflix/spinnaker/echo/api/events/NotificationAgent.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020 Armory, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.api.events;
+
+import com.netflix.spinnaker.kork.annotations.Alpha;
+import com.netflix.spinnaker.kork.plugins.api.internal.SpinnakerExtensionPoint;
+import java.util.Map;
+
+/** A NotificationAgent handles user-configured pipeline notifications. */
+@Alpha
+public interface NotificationAgent extends SpinnakerExtensionPoint {
```

Comment:
Would like to see `@Nonnull` etc on parameters and return types (or `@NonnullByDefault` etc)